### DISCUSSION TITLE
How do I change bun package manager cache in Windows?

### DISCUSSION THREAD

**Discussion by juliomuhlbauer** [oven-sh/bun#11981] [Q&A]
I want to move the cache from the C disk to the F disk

**Comment by Jarred-Sumner** [comment]
You could set the environment variable `BUN_INSTALL_CACHE_DIR`

it's supposed to be smart enough to do this automatically though

**Reply by ZeroAurora** [reply]
I didn't find this environment variable anywhere in documentation, should we add this (and more envs) to the docs?
And you said "it's supposed to be smart enough to do this automatically though", but what I experience is that all caches goes to `$USERPROFILE$\.bun` . Is it an expected behavior?

Edit: tried again and works. you need to write your cache location like `D:\\some\\location`

**Comment by Super1Windcloud** [comment]
where  is on document   ?

**Comment by Super1Windcloud** [comment]
![image](https://github.com/user-attachments/assets/ebb9eb6f-1fd5-4f37-bb9c-dcd03ba30a70)


how to change cache dir

**Reply by donywan** [reply]
I use `$USERPROFILE$\.bunfig.toml` but can not change.

**Reply by Super1Windcloud** [reply]
> I use `$USERPROFILE$\.bunfig.toml` but can not change.我使用`$USERPROFILE$\.bunfig.toml`但无法更改。

yes  , me  to,  So  don't consider bun  currently

**Reply by fardarter** [reply]
I've set up a `.bunfig.toml` file with the following configuration but can't seem to change the cache location. Trying to put things in one folder for pipeline caching.

```
# Bunfig docs: https://bun.sh/docs/runtime/bunfig

[install]
auto = "auto"
globalDir = "./build/cache/.bun/install/global"
globalBinDir = "./build/cache/.bun/bin"

[install.cache]
# the directory to use for the cache
dir = "./build/cache/.bun/install/cache"

[install.lockfile]
save = true
print = "yarn"
```

**Reply by donywan** [reply]
Like this.
```
[install]

# where `bun install --global` installs packages
globalDir = "d:\\.bun\\install\\global"

# where globally-installed package bins are linked
globalBinDir = "d:\\.bun\\bin"

# set default registry as a string
registry = "https://registry.npmmirror.com/"

[install.cache]

# the directory to use for the cache
dir = "d:\\.bun\\install\\cache"

# when true, don't load from the global cache.
# Bun may still write to node_modules/.cache
disable = false

# when true, always resolve the latest versions from the registry
disableManifest = false

```

**Reply by fardarter** [reply]
@donywan Could you help me understand the significance of the differences here?

**Reply by donywan** [reply]
You can try using an absolute path.

**Comment by dkebler** [comment]
on linux, here is mine,  `globalBinDir` gets the links but `globalDir` and the cache directory are ignored and they are put in the default of  ~/.bun.   Is this a bug??  Why are these settings ignored?

```
[install]
# optional = true
# dev = true
# peer = true
# production = false
# exact = false
# auto = "auto"
# frozenLockfile = false
# dryRun = false
globalDir = "/opt/bun/global"
globalBinDir = "/opt/bin"

[install.cache]
dir = "/tmp/cache/bun"
# disable = false
# disableManifest = false

# [install.lockfile]
# save = true
```
